components/formats-gpl/src/loci/formats/in/SlideBook6Reader.java

```diff
@@ -0,0 +1,393 @@
+
+	// -- Static initializers --
+
+	private static boolean libraryFound = false;
+
+	static {
+		String library_path = System.getProperty("java.library.path");
+
+		try {
+			// load JNI wrapper of SBReadFile.dll
+			System.loadLibrary("SlideBook6Reader");
+			libraryFound = true;
+		}
+		catch (UnsatisfiedLinkError e) {
+			LOGGER.warn(NO_3I_MSG, e);
+			libraryFound = false;
+		}
+		catch (SecurityException e) {
+			LOGGER.warn("Insufficient permission to load native library", e);
+			libraryFound = false;
+		}
+	}
+
+	// -- Constructor --
+
+	public SlideBook6Reader() {
+		super("SlideBook 6 SLD (native)", new String[] {"sld"});
+		domains = new String[] {FormatTools.LM_DOMAIN};
+		suffixSufficient = false;
+	}
+
+	// -- IFormatReader API methods --
+
+	/* @see loci.formats.IFormatReader#isThisType(RandomAccessInputStream) */
+	public boolean isThisType(RandomAccessInputStream stream) throws IOException {
+		final int blockLen = 8;
+		stream.seek(4);
+		boolean littleEndian = stream.readString(2).equals("II");
+		if (!FormatTools.validStream(stream, blockLen, littleEndian)) return false;
+		int magicBytes1 = stream.readShort();
+		int magicBytes2 = stream.readShort();
+
+		boolean isMatch = ((magicBytes2 & 0xff00) == SLD_MAGIC_BYTES_1_1 ||
+				(magicBytes2 & 0xff00) == SLD_MAGIC_BYTES_1_2) &&
+				(magicBytes1 == SLD_MAGIC_BYTES_1_0 ||
+				magicBytes1 == SLD_MAGIC_BYTES_2_0);
+
+		return isMatch;
+	}
+
+	/**
+	 * @see IFormatReader#isThisType(String, boolean)
+	 */
+	public boolean isThisType(String file, boolean open) {
+		// Check the first few bytes to determine if the file can be read by this reader.
+		return libraryFound && super.isThisType(file, open);
+	}
+
+	/**
+	 * @see loci.formats.IFormatReader#openBytes(int, byte[], int, int, int, int)
+	 */
+	public byte[] openBytes(int no, byte[] buf, int x, int y, int w, int h)
+			throws FormatException, IOException
+	{
+		FormatTools.checkPlaneParameters(this, no, buf.length, x, y, w, h);
+
+		int[] zct = FormatTools.getZCTCoords(this, no);
+		int bpc = FormatTools.getBytesPerPixel(getPixelType());
+		byte[] b = new byte[FormatTools.getPlaneSize(this)];
+
+		readImagePlaneBuf(b, getSeries(), 0, zct[2], zct[0], zct[1]);
+
+		int pixel = bpc * getRGBChannelCount();
+		int rowLen = w * pixel;
+		for (int row=0; row<h; row++) {
+			System.arraycopy(b, pixel * ((row + y) * getSizeX() + x), buf,
+					row * rowLen, rowLen);
+		}
+
+		if (isRGB()) {
+			int bpp = getSizeC() * bpc;
+			int line = w * bpp;
+			for (int row=0; row<h; row++) {
+				for (int col=0; col<w; col++) {
+					int base = row * line + col * bpp;
+					for (int bb=0; bb<bpc; bb++) {
+						byte blue = buf[base + bpc*(getSizeC() - 1) + bb];
+						buf[base + bpc*(getSizeC() - 1) + bb] = buf[base + bb];
+						buf[base + bb] = blue;
+					}
+				}
+			}
+		}
+		return buf;
+	}
+
+	// -- Internal FormatReader API methods --
+	public void close(boolean fileOnly) throws IOException {
+		super.close(fileOnly);
+		closeFile();
```

Comment:
The call to ```closeFile()``` needs to be in an ```if (libraryFound)``` block.  ```close(fileOnly)``` can be called on an uninitialized reader, which in this case means that an ```UnsatisfiedLinkError``` will be thrown when the .dlls are not present even if this reader isn't being used.